### ISSUE TITLE
subHour stops working in non-UTC timezones

### ISSUE THREAD

**Issue by jakenoble** [briannesbitt/Carbon#1188]
`$to` is a Carbon object with New York Timezone set to yesterday at 23:59:59
`$from` is a Carbon object with New York Timezone set to yesterday at 00:00:00

Therefore `$from` is the start of yesterday and `$to` is at the end of yesterday.

```
for ($date = $to->copy(); $date->greaterThan($from); $date->subHour()) {
    // Do something
}
```

This loop should go through each hour of the day, it does this if I set the timezone to 'UTC'. 

But, if its set the timezone something else (e.g. New York) then `subHour()` stops working at 03:59:59 and the loop will go on forever. This is because `$date` is no longer moving towards the `$from` date. And so `$date->greaterThan($from)` is never true.

Is this a bug? Or is this expected, for some reason, when using timezones?

The above issue does not occur if the dates are something other than yesterday.

**Comment by kylekatarnls** [comment]
Confirmed. I'll check.

**Comment by kylekatarnls** [comment]
Meanwhile for working on 1.24, I suggest you:

```php
for ($date = $to->copy(); $date->greaterThan($from); $date->setTimestamp($date->getTimestamp() - Carbon::SECONDS_PER_MINUTE * Carbon::MINUTES_PER_HOUR)) {
    echo $date->format('H:i')."\n";
}
```

**Comment by jakenoble** [comment]
@kylekatarnls thanks a lot!

**Comment by kylekatarnls** [comment]
`subRealHour` will be available in the next version via #1208 so I close this issue.
